admin/acceptance.py

```diff
@@ -950,6 +951,101 @@ def eliot_output(message):
     sys.stdout.write(format % message)
     sys.stdout.flush()
 
+def capture_upstart(reactor, host, output_file):
+    """
+    SSH into given machine and capture relevant logs, writing them to
+    output file.
+
+    :param reactor: The reactor.
+    :param bytes host: Machine to SSH into.
+    :param file output_file: File to write to.
+    """
+    # TODO: check the format we need to use here (if we need it at all)
+    # can use it later. Using the whole filename instead of smtg like flocker*
+    # because I can't make it work inside the command. It has a funny way of
+    # parsing the commands. Suggestions are welcome :D
+    formatter = tail_formatter(output_file, host)
+    ran = run_ssh(
+        reactor=reactor,
+        host=host,
+        username='root',
+        command=[
+            b'tail',
+            b'/var/log/flocker/flocker-control.log',
+            b'/var/log/flocker/flocker-dataset-agent.log',
+            b'/var/log/flocker/flocker-container-agent.log',
+            b'/var/log/flocker/flocker-docker-plugin.log',
+            b'/var/log/upstart/docker.log',
+        ],
+        handle_stdout=formatter.handle_output_line,
+    )
+    ran.addErrback(write_failure, logger=None)
+    # Deliver a final empty line to process the last message
+    ran.addCallback(lambda ignored: formatter.handle_output_line(b""))
+
+
+class tail_formatter(object):
+    """
+    Formatter for the output of the ``tail`` commands that will produce logs
+    with Eliot messages with the same format as the ones produced when
+    parsing journalctl output
+
+    :ivar file output_file: log file where we want to write our log
+    :ivar bytes _host: ip address or identifier of our host to be
+        added to the Elliot messages
+    :ivar bytes service: optional initial name of the service. This initial
+        name shouldn't appear anywhere unless there is an error, as the first
+        line of the output of trial will be a file name, that will be used
+        to set the name of the service we are currently parsing
+    """
+    def __init__(self, output_file, host, service = "unknown"):
+        self._output_file = output_file
+        self._host = host
+        self._service = service
+        self.service_regexp = re.compile(r"[/var/log/flocker/]|[/log/upstart/](.*)\.log")
```

Comment:
I don't think this regex is exactly right.

```
In [1]: import re

In [2]: some_re = re.compile(r"[/var/log/flocker/]|[/log/upstart/](.*)\.log")

In [3]: some_re.search('v.log')
Out[3]: <_sre.SRE_Match at 0x1048fae40>
```

The square brackets indicate character classes. In order to match literal `[` and `]`, we need to escape them with `\`.